### PR TITLE
JS module for uploading and fetching object URLs in Linode Object Storage

### PR DESCRIPTION
**Pull request by Ushmajit**
Tested the module by writing a sample node starter file and then resuing the module.

Expected behavior verified :

1. Verified that file is being uploaded successfully into the bucket.
2. Verified object Url is properly fetched using Linode V4 APIs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "dependencies": {
+    "aws-sdk": "^2.820.0",
+    "node-fetch": "^3.0.0",
+    "path": "^0.12.7",
+    "url-join": "^4.0.1",
+    "winston": "^3.3.3"
```
Comment (abose) [inline package.json:14]: unused?

**Review by abose — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "author": "ushmajit",
+  "type": "module",
+  "license": "ISC"
```
Comment (abose) [inline package.json:18]: GNU AGPL

File: src/linode_object_storage_module.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * AWS S3 Client. Please refer to https://docs.aws.amazon.com/sdk-for-javascript/index.html for more details.
+ *
+ * @param accessKeyId bucket specific unique identifier required for authentication
+ * @param secretAccessKey user specific unique identifier required for authentication
+ * @param region indicates the geographical server location (e.g us-east-1, eu-west-1a)
+ * @param file complete path of the file to be uploaded is passed on as a parameter
+ * @param bucket uniquely identifies the bucket where the file should be uploaded
+ * @returns {Promise<void>}
+ */
+async function uploadFileToLinodeBucket (accessKeyId, secretAccessKey, region, file, bucket) {
```
Comment (abose) [inline src/linode_object_storage_module.js:18]: Would be nice to have some predefined constants for regions

**Comment by abose** [comment]
Unit test notes:
While we are starting to choose JS unit test frameworks, just some notes
* [Jest](https://jestjs.io/) is a good framework for unit tests. It has straightforward integration to test react/angular/stuff and is maintained by FB.
* [Mocha](https://mochajs.org/) is used by brackets/Phoenix, so it might be good to have mocha for uniform test framework across our repos. Also has been around for a decade and is widely used.

**Comment by abose** [comment]
* https://github.com/aicore/template-nodejs-ts is a template repo that can be used to quickly create nodejs project scaffolding. But it is still in development :package: 
* If there is something that is built in this lib that can be shared to the template, consider adding it there too for future pulls.
* Also there are some issue/pr template files that are in the template but needs to be copied to this repo.

**Comment by abose** [comment]
Done. I have updated https://github.com/aicore/template-nodejs with best practices we follow for brackets and phoenix.
* Integrated unit and integration test frameworks
* Integrated static code analytics tools
* github actions, git commit taps etc.

Just copy and merge the contents of the repo here to get all the functions. Follow readme.  :)

File: src/linode_object_storage_module.js (hunk 3 above)
Comment (Ushmajit) [reply src/linode_object_storage_module.js:18]: Sure, makes sense, created an issue to track this https://github.com/aicore/linode-object-storage-lib/issues/3

File: package.json (hunk 1 above)
Comment (Ushmajit) [reply package.json:14]: My bad removed unused lib Winston.

File: package.json (hunk 2 above)
Comment (Ushmajit) [reply package.json:18]: Changed

**Comment by Ushmajit** [comment]
Thanks for the template files, copied all the relevant files from the template to the module. This is the new PR https://github.com/aicore/linode-object-storage-lib/pull/2

**Pull request merged**